console/src/components/Common/utils/jsUtils.tsx

```diff
@@ -353,3 +353,40 @@ export const getCurrTimeForFileName = () => {
 export const convertDateTimeToLocale = (dateTime: string) => {
   return moment(dateTime, moment.ISO_8601).format('ddd, MMM Do HH:mm:ss Z');
 };
+
+export const prioritySearch = (
+  searchText: string,
+  searchList: Record<string, any>[],
+  searchProp: string
+) => {
+  let searchResults: Record<string, any>[] = [];
+  if (searchText) {
+    const primarySameCaseResults: Record<string, any>[] = [];
+    const secondarySameCaseResults: Record<string, any>[] = [];
+    const primaryOppCaseResults: Record<string, any>[] = [];
+    const secondaryOppCaseResults: Record<string, any>[] = [];
+    searchList.forEach(term => {
+      const currentTerm = term[searchProp];
+      if (currentTerm.startsWith(searchText)) {
+        primarySameCaseResults.push(term);
+      } else if (currentTerm.includes(searchText)) {
+        secondarySameCaseResults.push(term);
+      } else if (
+        currentTerm.toLowerCase().startsWith(searchText.toLowerCase())
+      ) {
+        primaryOppCaseResults.push(term);
+      } else if (currentTerm.toLowerCase().includes(searchText.toLowerCase())) {
+        secondaryOppCaseResults.push(term);
+      }
+    });
+    searchResults = [
+      ...primarySameCaseResults,
+      ...secondarySameCaseResults,
+      ...primaryOppCaseResults,
+      ...secondaryOppCaseResults,
+    ];
+  } else {
+    searchResults = [...searchList];
+  }
```

Comment:
In my opinion, I feel it is better to handle these cases, before coming to the main part of the function. So my suggestion would be to just initialize the variable `searchResults` with `searchList` itself and eliminate this `else` case

Reply:
Yeah. Got it! Will make the changes.